type_decoders/README.md

```diff
@@ -0,0 +1,787 @@
+
+### isInteger()
+
+```ts
+interface IIntegerDecoderOptions {
+  msg?: DecoderErrorMsgArg;
+}
+
+function isInteger(options?: IIntegerDecoderOptions): Decoder<number, unknown>;
+```
+
+`isInteger()` can be used to verify that an unknown value is a whole `number`.
+
+### isUndefined()
+
+```ts
+interface IUndefinedDecoderOptions {
+  msg?: DecoderErrorMsgArg;
+}
+
+function isUndefined(options?: IUndefinedDecoderOptions): Decoder<undefined, unknown>;
+```
+
+`isUndefined()` can be used to verify that an unknown value is `undefined`. This is a convenience function for `isExactly(undefined)`.
+
+### isNull()
+
+```ts
+interface INullDecoderOptions {
+  msg?: DecoderErrorMsgArg;
+}
+
+function isNull(options?: INullDecoderOptions): Decoder<null, unknown>;
+```
+
+`isNull()` can be used to verify that an unknown value is `null`. This is a convenience function for `isExactly(null)`.
+
+### isRegex()
+
+```ts
+interface IRegexDecoderOptions {
+  msg?: DecoderErrorMsgArg;
+}
+
+function isRegex(regex: RegExp, options?: IRegexDecoderOptions): Decoder<string, unknown>;
+```
+
+`isRegex()` can be used to verify that an unknown value is `string` which conforms to the given `RegExp`.
+
+### isAny()
+
+```ts
+function isAny<T = unknown>(): Decoder<T, unknown>;
+```
+
+`isAny()` creates a decoder which always returns `DecoderSuccess` with whatever input value is provided to it.
+
+### isNever()
+
+```ts
+interface INeverDecoderOptions {
+  msg?: DecoderErrorMsgArg;
+}
+
+function isNever(options?: INeverDecoderOptions): Decoder<never, unknown>;
+```
+
+`isNever()` creates a decoder which always returns `DecoderError` with whatever input value is provided to it. One use case is using it in combination with `isObject` and `isOptional` to assert that an input object doesn't contain a given key.
+
+Example:
+
+```ts
+const validator = assert(isObject({a: isString() b: isOptional(isNever()) }));
+
+validator({ a: 'one' }) // returns { a: 'one' }
+validator({ a: 'one', b: 'two' }) // throws DecoderError
+```
+
+### isExactly()
+
+```ts
+interface IExactlyDecoderOptions {
+  msg?: DecoderErrorMsgArg;
+}
+
+function isExactly<T>(value: T, options?: IExactlyDecoderOptions): Decoder<T, unknown>;
+```
+
+`isExactly()` accepts a `value: T` argument and can be used to verify that an unknown input is `=== value`.
+
+### isConstant()
+
+```ts
+function isConstant<T>(value: T): Decoder<T, unknown>;
+```
+
+`isConstant()` accepts a `value: T` argument and creates a decoder which always returns the `value: T` argument, ignoring its input value.
+
+### isInstanceOf()
+
+```ts
+interface IInstanceOfDecoderOptions {
+  msg?: DecoderErrorMsgArg;
+}
+
+function isInstanceOf<T extends new (...args: any) => any>(clazz: T, options?: IInstanceOfDecoderOptions): Decoder<InstanceType<T>, unknown>;
+```
+
+`isInstanceOf()` accepts a javascript constructor argument and creates a decoder which verifies that its input is `instanceof clazz`.
+
+### isCheckedWith()
```

Comment:
`isMatchForPredicate`?

If we got rid of the `is` prefix, this could be `matchesPredicate`.